src/core/Akka/Actor/ActorRefProvider.cs

```diff
@@ -296,7 +296,7 @@ private RootGuardianActorRef CreateRootGuardian(ActorSystemImpl system)
                 return Directive.Stop;
             });
             var props = Props.Create<GuardianActor>(rootGuardianStrategy);
-            var rootGuardian = new RootGuardianActorRef(system, props, DefaultDispatcher, _defaultMailbox, supervisor, _rootPath, _deadLetters, _extraNames);
+            var rootGuardian = new RootGuardianActorRef(system, props, InternalDispatcher, _defaultMailbox, supervisor, _rootPath, _deadLetters, _extraNames);
```

Comment:
So the only reason I haven't merged this PR yet - we are now guaranteed to have to create threads in the `InternalDispatcher` tool each and every time we create an `ActorSystem`, even for these `RootGuardianActorRef` actors who are very infrequently used.... I need to check the `DedicatedThreadPool` code to see if we allow those threads to terminate if there's not enough work for them to do. That'd be ideal and a better use of CPU resources rather than having those threads idly polling for work all the time.

Reply:
Did this check - concluded that it is pretty safe.